### PR TITLE
AB#87701_ABC-see-if-possible-to-deactivate-cross-page-filtering-so-each-page-handles-its-own-filter

### PR DESCRIPTION
**Pull request by unai-reliefapp**
# Description

- feat: avoid merge of current context filter value on survey first load for web-components 
- feat: disable share filters feature for dashboards in web-widgets

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/87701)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please refer to video below

## Screenshots

https://github.com/ReliefApplications/ems-frontend/assets/123092672/d2562eeb-220c-4365-8fac-fc440bd561a7

# Checklist:

( * == Mandatory ) 

- [x] * I have set myself as assignee of the pull request
- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [ ] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

File: libs/shared/src/lib/components/dashboard-filter/dashboard-filter.component.ts [hunk 1]
```diff
@@ -320,15 +323,20 @@ export class DashboardFilterComponent
...
-    filterKeys
-      .filter((key) => !currentSurveyQuestions.find((sq) => sq.name === key))
-      .forEach((key) => {
-        nextFilterValue[key] = currentFilterValue[key];
-      });
+    // Don't merge current context filter values to the filter if survey has init and it's used in web component
+    if (
+      !(this.surveyInit && this.contextService.shadowDomService.isShadowRoot)
+    ) {
+      const currentFilterValue = this.contextService.filter.value;
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/dashboard-filter/dashboard-filter.component.ts]: @unai-reliefapp 
it looks like there you're still doing cross page filtering, as the value is coming from the service
Is it correct?

and the service is not provided by dashboard, so all dashboards are using the same service instance

Comment (unai-reliefapp) [reply libs/shared/src/lib/components/dashboard-filter/dashboard-filter.component.ts]: They are sharing the service but the filter would be trigger only for each dashboard with their own values.
On first load of any dashboard the filters are set with the related dashboard filters default value only,
And then once is loaded, the filter is triggered with the value of the current survey, therefor even if it's the same service, the values that it handles are only the values of the current loaded dashboard filter for the current loaded dashboard

KR, Unai

Comment (AntoineRelief) [reply libs/shared/src/lib/components/dashboard-filter/dashboard-filter.component.ts]: @unai-reliefapp 
okay! Thanks for the clarification
I'll test that

Comment (unai-reliefapp) [reply libs/shared/src/lib/components/dashboard-filter/dashboard-filter.component.ts]: > @unai-reliefapp okay! Thanks for the clarification I'll test that

Welcome,
If you think of another way that would be better to implement the filtering for each page let me know

KR, Unai

Comment (AntoineRelief) [reply libs/shared/src/lib/components/dashboard-filter/dashboard-filter.component.ts]: @unai-reliefapp indeed, working as expected, thanks unai 👍

Comment (AntoineRelief) [reply libs/shared/src/lib/components/dashboard-filter/dashboard-filter.component.ts]: I'll now wait for their confirmation and merge it

**Comment by unai-reliefapp** [comment]
@AntoineRelief Integration of shared filters with an env property ready 👍 , waiting for the case of how to handle region selection on set country value
Previous code is commented
Video attached

**Comment by unai-reliefapp** [comment]
@AntoineRelief I think this can be closed

**Pull request closed without merging**